tests/cbmc/proofs/Makefile.common

```diff
@@ -0,0 +1,279 @@
+# Don't execute deep checks by default
+AWS_DEEP_CHECKS ?= 0
+DEFINES += -DAWS_DEEP_CHECKS=$(AWS_DEEP_CHECKS)
+
+################################################################
+# Remove function bodies that are not used in the proofs. 
+# Removing the function from the goto program helps CBMC's
+# function pointer analysis
+
+# We override abort() to be assert(0)
+ABSTRACTIONS += $(HELPERDIR)/stubs/abort_override_assert_false.c
+REMOVE_FUNCTION_BODY += --remove-function-body abort
+
+# Users can remove additonal functions using this paramater
+REMOVE_FUNCTION_BODY += $(ADDITIONAL_REMOVE_FUNCTION_BODY)
+
+################################################################
+# Determine the names of the goto files to build
+#DEPENDENT_GOTOS = $(call GOTO_FROM_C,$(DEPENDENCIES))
+ENTRY_GOTO = $(GOTODIR)/$(ENTRY)
+
+################################################################
+# Temporarily disable UNWIND_GOTO, SIMPLIFY until the feature is
+# stable
+UNWIND_GOTO ?= 0
+SIMPLIFY ?= 0
+
+################################################################
+# Build targets that make the relevant .goto files
+
+# Other libraries that use this Makefile template may wish to do
+# actual work here
+setup_dependencies:
+	echo "Setting up dependencies"
+
+# Here, whenever there is a change in any of ANSI-C source
+# dependency files, make will take action. However, to make
+# sure changes in the headers files will also trigger make,
+# the user must run make clean first.
+$(ENTRY_GOTO)0.goto: setup_dependencies $(ENTRY).c $(DEPENDENCIES)
+	mkdir -p $(dir $@)
+	mkdir -p $(dir $(call LOG_FROM_ENTRY,$@))
+	$(call DO_GOTO_CC,\
+		--export-function-local-symbols \
+			$(CBMC_VERBOSITY) \
+			--function $(ENTRY) \
+			$(INC) \
+			$(DEFINES), \
+		$(ENTRY).c $(DEPENDENCIES),\
+		$@)
+
+# Removes specified function bodies. This allows us to replace
+# function definitions with ABSTRACTIONS.
+$(ENTRY_GOTO)1.goto: $(ENTRY_GOTO)0.goto
+ifeq ($(REMOVE_FUNCTION_BODY),"")
+	$(call DO_NOOP_COPY,"Not removing function bodies",$<,$@)
+else
+	$(call DO_GOTO_INSTRUMENT,$(REMOVE_FUNCTION_BODY),$<,$@)
+endif
+
+# ABSTRACTIONS is a list of function stubs to use. If a function body
+# is missing and is not abstracted, then it returnes a non
+# deterministic value.
+$(ENTRY_GOTO)2.goto: $(ENTRY_GOTO)1.goto
+ifeq ($(ABSTRACTIONS),"")
+	$(call DO_NOOP_COPY,"Not implementing abstractions",$<,$@)
+else
+	$(call DO_GOTO_CC,--function $(ENTRY) $(ABSTRACTIONS) $(INC) $(DEFINES),$<,$@)
+endif
+
+# Simplify and constant propagation may benefit from unwinding first
+$(ENTRY_GOTO)3.goto: $(ENTRY_GOTO)2.goto
+ifeq ($(UNWIND_GOTO),1)
+	$(call DO_GOTO_INSTRUMENT,$(UNWINDING),$<,$@)
+else
+	$(call DO_NOOP_COPY,"Not unwinding goto program",$<,$@)
+endif
+
+# Skip simplify (and hence generate-function-body) until missing source locations debugged
+$(ENTRY_GOTO)4.goto: $(ENTRY_GOTO)3.goto
+ifeq ($(SIMPLIFY),1)
+	$(call DO_GOTO_INSTRUMENT,--generate-function-body '.*',$<,$@)
```

Comment:
I have no idea what `'.*'` expands to...but if this whole rule isn't necessary,  please remove it

Reply:
Turns out its a regular expression.  Will make a comment